### PR TITLE
Update media3-backend.md docs

### PR DESCRIPTION
**Pull request by yschimke**
#### WHAT

Update media3-backend.md docs, document Audio Offload, Bluetooth Connections, and Logging.

#### WHY

Have somewhere to point to in docs rather than code.

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: docs/media3-backend.md [hunk 1]
```diff
@@ -4,6 +4,91 @@
 
 For more information, visit the documentation: https://google.github.io/horologist/media3-backend
```
Comment (luizgrp) [inline docs/media3-backend.md:5]: nit: should remove this line as it's pointing to itself

File: docs/media3-backend.md [hunk 2]
```diff
@@ -4,6 +4,91 @@
...
+These build on top of the [Media3](https://developer.android.com/jetpack/androidx/releases/media3)
+player featuring ExoPlayer which is optimised for Wear Playback, and is the standard playback
+engine for Wear media apps.
+
+All functionality is demonstrated in the `media-sample` app, and as such is not described here
+with extensive code samples.
+
+### Bluetooth Headphone Connections
+
+Any extended playback such as music, podcasts, or radio should only use a connected bluetooth
```
Comment (luizgrp) [inline docs/media3-backend.md:22]: nit: could link this orientation to this https://developer.android.com/training/wearables/principles#bluetooh-headphones

**Review by luizgrp — APPROVED** [review]
Nice, thanks!

File: docs/media3-backend.md (hunk 2 above)
Comment (yschimke) [reply docs/media3-backend.md:22]: Nice

**Pull request merged**